cranelift-codegen/meta/src/cdsl/operands.rs

```diff
@@ -74,6 +74,15 @@ impl Operand {
             _ => false,
         }
     }
+
+    /// Determine the name to use for this operand value in the generated Rust code.
+    pub fn rust_name(&self) -> &'static str {
+        if let Some(name) = self.kind.default_member {
+            name
+        } else {
+            self.name
+        }
+    }
```

Comment:
@bnjbvr: is this correct? This change uses the `self.kind.default_member` branch of the `if-else` but I don't know if the `else` is correct (otherwise `self.kind.name`?).

Reply:
Good question; at some point I'll try to clean up all the "name", because they're not very eloquent. In this case, I think we can use `kind.imm_name()` in the caller, and not add the new function; can you check this, please? If not, then since we're really asking for the operand kind *member* name, it should be `self.kind.name`.

Reply:
I tried `InstructionPredicate::new_is_field_equal(..., immediate_operand.kind.imm_name().unwrap(), ..)` but this failed with:
```
thread 'main' panicked at 'unexpected immediate field named intcc in instruction format IntCompare', cranelift-codegen/meta/src/cdsl/formats.rs:77:17
stack backtrace:
```
I think the issue here is that `predicates::is_equal` needs to use the actual Rust name for the IntCC value (https://github.com/CraneStation/cranelift/blob/d2feec1bb5854ca03f9fcdda6d58209c3acbd124/cranelift-codegen/meta/src/cdsl/instructions.rs#L827). And I think this is generated from `default_member` if it is set. In the cases where `default_member` is not set, I think (but am not sure) that the generator uses `Operand::name` and not `OperandKind::name`, which would make sense because you want to use names in Rust like `y` or `x` and not `intcc` and `uint8` that could be duplicated in certain formats.

Reply:
Right (definitely needs refactoring :)). So the default member is the operand's name when it's not defined, and it's not defined only for variable arguments. We can just use `default_member` here and `expect` with an error message in the caller.